src/resources/js/tickets-registration-page.js

```diff
@@ -1,351 +1,355 @@
+		fieldsErrorRequired: '.tribe-block__tickets__item__attendee__fields__error--required',
+		fieldsErrorAjax: '.tribe-block__tickets__item__attendee__fields__error--ajax',
+		fieldsSuccess: '.tribe-block__tickets__item__attendee__fields__success',
+		loader: '.tribe-block__tickets__item__attendee__fields__loader',
+		form: '.tribe-block__tickets__item__attendee__fields__form',
+		toggler: '.tribe-block__tickets__registration__toggle__handler',
+		status: '.tribe-block__tickets__registration__status',
+		field: {
+			text: '.tribe-block__tickets__item__attendee__field__text',
+			checkbox: '.tribe-block__tickets__item__attendee__field__checkbox',
+			select: '.tribe-block__tickets__item__attendee__field__select',
+			radio: '.tribe-block__tickets__item__attendee__field__radio',
+		},
+		checkout: '.tribe-block__tickets__registration__checkout',
+		checkoutButton: '.tribe-block__tickets__registration__checkout__submit'
+	};
+
+	var $tribe_registration = $( obj.selector.container );
+
+	// Bail if there are no tickets on the current event/page/post
+	if ( 0 === $tribe_registration.length ) {
+		return;
+	}
+
+	/**
+	 * Handle the toggle for each event
+	 *
+	 * @since 4.9
+	 *
+	 * @return void
+	 */
+	$( obj.selector.container ).on(
+		'click',
+		obj.selector.toggler,
+		function( e ) {
+			e.preventDefault();
+
+			var $this = $( this );
+			var $event = $this.closest( obj.selector.container);
+
+			$event.find( obj.selector.fields ).toggle();
+			$this.toggleClass( 'open' );
+
+		} );
+
+	/**
+	 * Check if the required fields have data
+	 *
+	 * @since 4.9
+	 *
+	 * @return void
+	 */
+	obj.validateEventAttendees = function( $form ) {
+		var isValid = true;
+		var $fields = $form.find( '.tribe-tickets-meta-required' );
+
+		$fields.each( function() {
+			var $field = $( this );
+			var val = '';
+
+			if (
+				$field.is( obj.selector.field.radio ) ||
+				$field.is( obj.selector.field.checkbox )
+			) {
+				val = $field.find( 'input:checked' ).length ? 'checked' : '';
+			} else if ( $field.is( obj.selector.field.select ) ) {
+				val = $field.find( 'select' ).val();
+			} else {
+				val = $field.find( 'input, textarea' ).val().trim();
+			}
+
+			if ( 0 === val.length ) {
+				isValid = false;
+			}
+
+		} );
+
+		return isValid;
+	};
+
+	/**
+	 * Update container status to complete
+	 *
+	 * @since 4.10.1
+	 *
+	 * @return void
+	 */
+	obj.updateStatusToComplete = function( $event ) {
+		var $statusObj = $event.find( obj.selector.status );
+		$statusObj.removeClass( 'incomplete' );
+		$statusObj.find( 'i' ).removeClass( 'dashicons-edit' );
+		$statusObj.find( 'i' ).addClass( 'dashicons-yes' );
+	};
+
+	/**
+	 * Update container status to incomplete
+	 *
+	 * @since 4.10.1
+	 *
+	 * @return void
+	 */
+	obj.updateStatusToIncomplete = function( $event) {
+		var $statusObj = $event.find( obj.selector.status );
+		$statusObj.addClass( 'incomplete' );
+		$statusObj.find( 'i' ).addClass( 'dashicons-edit' );
```

Comment:
If we are storing a var that needs to be reused, should we then store `$statusObj.find( 'i' )`?